src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator/help-checkoutCredentialsId.html

```diff
@@ -1,3 +1,15 @@
 <div>
-    <p>Credentials used to check out sources during a build.</p>
+    <p>
+        Credentials used to check out sources during a build.
+    </p>
+    <p>
+        The default choice of "Same as scan credentials" is sufficient so long as you don't
+        need Jenkins to push some changes back into these repositories.
+    </p>
+    <p>
+        Otherwise you should create an <strong>SSH username with private key</strong> credential and
```

Comment:
You could potentially push changes using your access token too. Just depends on what permissions you granted the token. Similarly, you might use HTTP(S) credentials normally for checkout, but using username/password, and want these to be distinct from the read-only access token used for scan credentials.

I do not want to speculate on what all the reasons there might be for preferring distinct credentials. There may be site policies about use of certain kinds of authentication for accessing company source code, etc.